code/hud/hudconfig.cpp

```diff
@@ -518,9 +516,32 @@ void hud_config_init_ui(bool API_Access, int x, int y, int w, int h)
 	}
 
 	if (!API_Access){
-		// add text
-		for(i=0; i<NUM_HUD_TEXT; i++){
-			HC_ui_window.add_XSTR(&HC_text[gr_screen.res][i]);
+		// Get our custom color names ready and add text
+		for (int i = 0; i < NUM_HUD_TEXT; i++) {
+			if (i >= 8 && i <= 10) {    // Check if it's a color index which are hardcoded to 8, 9, 10.
+				int colorIndex = 10 - i; // Calculate the index for HC_colors
```

Comment:
Wouldn't this reverse the indexes?  Index 10 becomes 0, 9 becomes 1, 8 becomes 2.  Not sure if this is desired or if i-8 is what you're looking for.

Reply:
Yes that's correct. This is because the defines are hardcoded in the order of green, blue, amber, but the actual button text defines are reverse order. So it needs to be flipped.